app/src/apdu_handler.c

```diff
@@ -113,6 +153,129 @@ bool process_chunk(volatile uint32_t *tx, uint32_t rx) {
+        case P1_ETH_FIRST:
+            tx_initialize_eth();
+            tx_reset();
+            extract_eth_path(rx, OFFSET_DATA);
+            // there is not warranties that the first chunk
+            // contains the serialized path only;
+            // so we need to offset the data to point to the first transaction
+            // byte
+            uint32_t path_len = sizeof(uint32_t) * hdPathLen;
+
+            // plus the first offset data containing the path len
+            data += path_len + 1;
+            if (len < path_len) {
+                THROW(APDU_CODE_WRONG_LENGTH);
+            }
+
+            // now process the chunk
+            len -= path_len + 1;
+            if (get_tx_rlp_len(data, len, &read, &to_read) != rlp_ok) {
+                THROW(APDU_CODE_DATA_INVALID);
+            }
+
+            // get remaining data len
+            max_len = saturating_add(read, to_read);
+            max_len = MIN(max_len, len);
+
+            added = tx_append(data, max_len);
+            if (added != max_len) {
+                THROW(APDU_CODE_OUTPUT_BUFFER_TOO_SMALL);
+            }
+
+            tx_initialized = true;
+
+            // if the number of bytes read and the number of bytes to read
+            //  is the same as what we read...
+            if ((saturating_add(read, to_read) - len) == 0) {
+                return true;
+            }
+            return false;
+        case P1_ETH_MORE:
+            if (!tx_initialized) {
+                THROW(APDU_CODE_TX_NOT_INITIALIZED);
+            }
+
+            uint64_t buff_len = tx_get_buffer_length();
+            uint8_t *buff_data = tx_get_buffer();
+
+            if (get_tx_rlp_len(buff_data, buff_len, &read, &to_read) !=
+                rlp_ok) {
+                THROW(APDU_CODE_DATA_INVALID);
+            }
+
+            uint64_t rlp_read = buff_len - read;
+
+            // either the entire buffer of the remaining bytes we expect
+            uint64_t missing = to_read - rlp_read;
+            max_len = len;
+
+            if (missing < len)
+                max_len = missing;
+
+            added = tx_append(data, max_len);
+
+            if (added != max_len) {
+                tx_initialized = false;
+                THROW(APDU_CODE_OUTPUT_BUFFER_TOO_SMALL);
+            }
+
+            // check if this chunk was the last one
+            if (missing - len == 0) {
+                tx_initialized = false;
+                return true;
+            }
+
+            return false;
+    }
+    THROW(APDU_CODE_INVALIDP1P2);
+}
+
+__Z_INLINE void handle_getversion(volatile uint32_t *flags, volatile uint32_t *tx, uint32_t rx) {
+    UNUSED(flags);
+    UNUSED(rx);
+#ifdef DEBUG
+    G_io_apdu_buffer[0] = 0xFF;
+#else
+    G_io_apdu_buffer[0] = 0;
+#endif
+    G_io_apdu_buffer[1] = LEDGER_MAJOR_VERSION;
+    G_io_apdu_buffer[2] = LEDGER_MINOR_VERSION;
+    G_io_apdu_buffer[3] = LEDGER_PATCH_VERSION;
+    G_io_apdu_buffer[4] = !IS_UX_ALLOWED;
```

Comment:
this line should be set to `0` since SDK won't reply any APDU message if the device is blocked. We don't need IS_UX_ALLOWED macro anymore